model/map/src/main/java/org/keycloak/models/map/client/MapClientProvider.java

```diff
@@ -92,18 +95,25 @@ public void updateClient() {
             /** This is runtime information and should have never been part of the adapter */
             @Override
             public Map<String, Integer> getRegisteredNodes() {
-                return clientRegisteredNodesStore.computeIfAbsent(entity.getId(), k -> new ConcurrentHashMap<>());
+                return Collections.unmodifiableMap(getMapForEntity()
```

Comment:
Do we need the `unmodifiableMap`  as `collect` is making a copy of the original map?

Reply:
I wanted to make sure that no caller if this interface would use it as a modifiable map and would assume that the results would be stored in the original map.
To see this in the tests and to avoid future confusion, I wrapped it as an unmodifiable Map, so that each modification would throw a runtime exception.